### PR TITLE
[Mac] NSButton that handles ICommands. Needed for Binding Dialog and …

### PR DESCRIPTION
**Pull request by CartBlanche**
…Variations PRs. Slight fix to CommandMenuItem too.

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+				this.command = value;
+				this.command.CanExecuteChanged += CanExecuteChanged;
+			}
+		}
+
+		public CommandButton ()
+		{
+			Activated += (object sender, EventArgs e) => {
+				this.command?.Execute (null);
```
Comment (Therzok) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs]: Avoid ref cycle caused by the lambda capture.

```suggestion
				var button = (CommandButton)sender;
				button.command?.Execute (null);
```

File: Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Windows.Input;
+using AppKit;
+
+namespace Xamarin.PropertyEditing.Mac.Controls.Custom
+{
+	public class CommandButton : NSButton
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs]:
```suggestion
	internal class CommandButton : NSButton
```

File: Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		public ICommand Command
+		{
+			get { return this.command; }
+			set
+			{
+				if (this.command != null)
+					this.command.CanExecuteChanged -= CanExecuteChanged;
+
+				this.command = value;
+				this.command.CanExecuteChanged += CanExecuteChanged;
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/CommandButton.cs]: Setting this to null shouldn't crash.

**Comment by CartBlanche** [comment]
@ermau Issues addressed and you've been reassigned.

**Pull request merged**